### PR TITLE
nixos/iso-image: add option for copytoram

### PR DESCRIPTION
**Pull request by Lassulus**
###### Motivation for this change

Add the possibility to boot the live stick with its contents copied into ram.

### CODE REVIEW

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 1]
```diff
@@ -50,6 +50,13 @@ let
     LINUX /boot/bzImage
     APPEND init=${config.system.build.toplevel}/init ${toString config.boot.kernelParams} nomodeset
     INITRD /boot/initrd
+
+    # A variant to boot with 'copytoram
```
Comment (bobvanderlinden) [inline nixos/modules/installer/cd-dvd/iso-image.nix]: `'copytoram` -> `'copytoram'`

**Comment by bobvanderlinden** [comment]
Thanks @Lassulus for the work on copytoram. This seems to me like an nice addition as well. I still need to give this a try.

I'm guessing the advantages are runtime performance and the ability to pull the USB/CD from the system once it has booted? What are your motivations to get this in?

**Comment by Lassulus** [comment]
@bobvanderlinden Thanks :)

I'm using it mainly for 2 reasons:
1) If I boot an usb image via drivedroid on android I want to be able to tether afterwards with my phone.
2) During nixos installation parties there are often not enough live mediums around for everyone to play around. With this change 1 live stick would suffice for everyone.

**Comment by bobvanderlinden** [comment]
> If I boot an usb image via drivedroid on android

Hah! That made my day. I'm the dev of DriveDroid 😄. Sounds like useful usecases to me. I'll give it a go right away.

**Comment by bobvanderlinden** [comment]
Works nicely on legacy bios:

```
qemu-system-x86_64 -m 4G --enable-kvm -cdrom $(nix-build -A iso_minimal.x86_64-linux.out nixos/release.nix)/iso/*.iso
```

And on EFI:

```
qemu-system-x86_64 -m 4G --enable-kvm -bios $(nix-build -A pkgs.OVMF)/FV/OVMF.fd -cdrom $(nix-build -A iso_minimal.x86_64-linux.out nixos/release.nix)/iso/*.iso
```

Code looks fine to me too :+1:

**Comment by Lassulus** [comment]
@bobvanderlinden it's a small world :D
thank you for reviewing

**Pull request merged**